Content.Shared/Humanoid/Markings/MarkingColoring.cs

```diff
@@ -0,0 +1,227 @@
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.List;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Humanoid.Markings;
+
+[Serializable, NetSerializable]
+public enum MarkingColoringType : byte
+{
+    SimpleColor,
+    SkinColor,
+    HairColor,
+    FacialHairColor,
+    AnyHairColor,
+    Tattoo,
+    EyeColor
+}
+
+
+[DataDefinition]
+[Serializable, NetSerializable]
+public sealed class MarkingColors
+{
+    /// <summary>
+    /// Coloring properties that will be used on any unspecified layer
+    /// </summary>
+    [DataField("default", true)]
+    public ColoringProperties Default { get; } = new();
+
+    /// <summary>
+    /// Layers with their own coloring type and properties
+    /// </summary>
+    [DataField("layers", true)]
+    public Dictionary<string, ColoringProperties>? Layers;
+}
+
+/// <summary>
+/// Properties for coloring. 
+/// </summary>
+[DataDefinition]
+[Serializable, NetSerializable]
+public sealed class ColoringProperties
+{
+    /// <summary>
+    /// Coloring type that will be used on that layer
+    /// </summary>
+    [DataField("type", true, required: true)]
+    public MarkingColoringType Type { get; } = MarkingColoringType.SkinColor;
```

Comment:
maybe making this nullable would be nicer instead of having a SimpleColor enum variant, to indicate that there is a lack of default color to select from within the character appearance data, and to instead use the Color field defined below